cfme/tests/generic_objects/test_instances.py

```diff
@@ -56,3 +115,127 @@ def test_generic_objects_crud(appliance, context, request):
 
         instance.delete()
         assert not instance.exists
+
+
+@pytest.mark.parametrize('context', [ViaUI])
+def test_generic_objects_crud_ui(appliance, context, request):
+    """
+        CRUD test for generic object via UI
+
+        Metadata:
+            test_flag: ui
+    """
+    with appliance.context.use(context):
+        definition = appliance.collections.generic_object_definitions.create(
+            name="rest_generic_class{}".format(fauxfactory.gen_alphanumeric()),
+            description="Generic Object Definition",
+            attributes={"addr01": "String"},
+            associations={"services": "Service"},
+            methods=["hello_world"])
+
+        with appliance.context.use(ViaREST):
+            instance = appliance.collections.generic_objects.create(
+                name='rest_generic_instance{}'.format(fauxfactory.gen_alphanumeric()),
+                definition=definition,
+                attributes={'addr01': 'Test Address'}
+            )
+            request.addfinalizer(instance.delete)
+        view = navigate_to(instance, 'Details')
+        assert view.is_displayed
+
+
+@pytest.mark.parametrize('button_group', [True, False],
+                         ids=['button_group_with_button', 'single_button'])
+@pytest.mark.parametrize('context', [ViaUI])
+def test_generic_objects_with_buttons_ui(appliance, definition, service, request, button_group,
+                                         context):
+    """
+        Tests buttons ui visibility assigned to generic object
+
+        Metadata:
+            test_flag: ui
+    """
+    with appliance.context.use(context):
+        myservice = MyService(appliance, name=service.name)
+        group_name = 'button_group_{}'.format(fauxfactory.gen_alphanumeric())
+        view = appliance.browser.create_view(BaseLoggedInPage)
+        if button_group:
+            definition.add_button_group(
+                name=group_name,
+                description='Group_button_description',
+                image='fa-user'
+            )
+            view.flash.assert_message(
+                'Custom Button Group "{}" has been successfully added.'.format(group_name))
+
+        button_name = 'button_{}'.format(fauxfactory.gen_alphanumeric())
+        definition.add_button(
+            name=button_name,
+            description='Button_description',
+            image='fa-home',
+            request=fauxfactory.gen_alphanumeric(),
+            button_group=group_name if button_group else None)
+        message = (
+            'Custom Button "{}" has been successfully added under the selected button group.'
```

Comment:
**Optional**
Looks like this is just used once, consider dropping it into the assert_message() call, then you don't need extra parenthesis to split the ternary statement onto multiple lines.